src/shared/watchdog.c

```diff
@@ -19,27 +19,30 @@ static char *watchdog_device = NULL;
 static usec_t watchdog_timeout = USEC_INFINITY;
 static usec_t watchdog_last_ping = USEC_INFINITY;
 
+static unsigned int usec_to_sec(usec_t val) {
+        usec_t t = DIV_ROUND_UP(val, USEC_PER_SEC);
+        return (unsigned int) t >= (UINT_MAX / 1000) ? (UINT_MAX / 1000) : t; /* Saturate to watchdog max */
```

Comment:
the UINT_MAX/1000 thing is kinda magic. Please add a 

```c
#define WATCHDOG_TIMEOUT_MAX_SEC (UINT_MAX/1000U)
```

or so somewhere, with a comment explaining this and a reference to the kernel 4.5 like you have it in the man page

Reply:
btw, we typically write just "unsigned" instead of "unsigned int"

Reply:
also, if the passed value is above UINT_MAX then this check will not work since you cast down to unsigned first, and then the comparison is too late

Reply:
Okay, I see. I think I have it right now.

Are you saying I should document the new maximum on the man page? It might be confusing since the actual value of UINT_MAX (and therefore the actual maximum watchdog timeout as well) is architecture dependant.